examples/register.rs

```diff
@@ -0,0 +1,17 @@
+use nakama_rs::*;
+use nakama_rs::api_client::ApiClient;
+
+fn main() {
+    let mut client = ApiClient::new("defaultkey", "127.0.0.1", config::DEFAULT_PORT, "http");
+    // Note that the minimum password length is 8 characters!
+    client.register("some@user.com", "password", "Username");
+
+    loop {
```

Comment:
`client.in_progress` returns false unless `client.tick` has been called before, because `client.tick` moves the request from `state.next_request` to `ongoing_request`.

We could adapt `in_progress` to return `self.state.borrow().next_request.is_some() || self.ongoing_request.is_some()` and then simplify the loop as `while client.in_progress() { client.tick() }`